plugins/Referrers/API.php

```diff
@@ -328,22 +330,88 @@ public function getUrlsFromWebsiteId($idSite, $period, $date, $idSubtable, $segm
      */
     public function getSocials($idSite, $period, $date, $segment = false, $expanded = false, $flat = false)
     {
-        $dataTable = Archive::createDataTableFromArchive(Archiver::WEBSITES_RECORD_NAME, $idSite, $period, $date, $segment, $expanded, false);
+        $dataTable = Archive::createDataTableFromArchive(Archiver::SOCIAL_NETWORKS_RECORD_NAME, $idSite, $period, $date, $segment, $expanded, $flat);
+
+        $dataTable->filter('ColumnCallbackAddMetadata', array('label', 'url', function ($name) {
+            return Social::getInstance()->getMainUrlFromName($name);
+        }));
+
+        $dataTable = $this->completeSocialTablesWithOldReports($dataTable, $idSite, $period, $date, $segment, $expanded, $flat);
+
+        $dataTable->filter('MetadataCallbackAddMetadata', array('url', 'logo', function ($url) { return Social::getInstance()->getLogoFromUrl($url); }));
+
+        return $dataTable;
+    }
+
+    private function completeSocialTablesWithOldReports($dataTable, $idSite, $period, $date, $segment, $expanded, $flat)
+    {
+        return $this->combineDataTables($dataTable, function() use ($idSite, $period, $date, $segment, $expanded, $flat) {
```

Comment:
Is that only needed for DataTableMaps? I see below in `combineDataTables ` we handle it also when it is a regular data table but not sure when this is needed?

Reply:
I see it is when data is stored the old way...